subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultNamedDomainObjectCollectionTest.groovy

```diff
@@ -31,15 +31,15 @@ class DefaultNamedDomainObjectCollectionTest extends AbstractNamedDomainObjectCo
 
     Instantiator instantiator = TestUtil.instantiatorFactory().decorateLenient()
 
-    final DefaultNamedDomainObjectCollection<Bean> container = new DefaultNamedDomainObjectCollection<Bean>(Bean, new IterationOrderRetainingSetElementSource<Bean>(), instantiator, namer, callbackActionDecorator)
-    final Bean a = new BeanSub1("a")
-    final Bean b = new BeanSub1("b")
-    final Bean c = new BeanSub1("c")
-    final Bean d = new BeanSub2("d")
-    final boolean externalProviderAllowed = true
+    DefaultNamedDomainObjectCollection<Bean> container = new DefaultNamedDomainObjectCollection<Bean>(Bean, new IterationOrderRetainingSetElementSource<Bean>(), instantiator, namer, callbackActionDecorator)
+    Bean a = new BeanSub1("a")
```

Comment:
Do they have any intention of fixing this? This sounds like a major breakage (esp given the ripple effect it has on buildscript authors)

Reply:
Ignore this, didn't realize these are methods :(

Reply:
I am now trying to come up with a minimal reproducer for this issue and I am getting failures for the same cases even with m4 which in our project did work.
For example:
```
class GetterClash extends spock.lang.Specification {
    final String bar = "BAR"
    String getBar() {
        return bar
    }
    def "some test"() {
        expect:
        true
    }
}
```
Fails with m4 (at test execution time):
```
java.lang.IllegalAccessError: Update to non-static final field GetterClash.bar attempted from a different method ($spock_initializeFields) than the initializer method <init> 
	at GetterClash.$spock_initializeFields(GetterClash.groovy)
```
Fails with m5 (at compile time and with a better message):
```
> Task :compileTestGroovy FAILED
startup failed:
/home/vaidotas/tmp/gradle/spock/m5/src/test/groovy/GetterClash.groovy: 6: Final field 'bar' conflicts with method 'getBar'; please rename either of them @ line 6, column 5.
       final String bar = "BAR"
```
In this case, removing the getter solves the "problem" (I did this in `AbstractIntegrationSpec` for `mavenRepo/ivyRepo` fields.


Same story with:
```
abstract class BaseSpec extends spock.lang.Specification {
    abstract String getFoo()
    def "base test"() {
        expect:
        true
    }
}
```
and
```
class ChildSpec extends BaseSpec {
    // this is effectively @Override int getFoo() { return 42 }
    // either being explicit that this is an override or removing final modifier solves the problem
    final String foo = "FOO";
    def "child test"() {
        expect:
        true
    }
}
```
With m4, I am seeing test failures:
```
java.lang.IllegalAccessError: Update to non-static final field ChildSpec.foo attempted from a different method ($spock_initializeFields) than the initializer method <init> 
	at ChildSpec.$spock_initializeFields(ChildSpec.groovy)
```
While m5 fails me at compile time:
```
> Task :compileTestGroovy FAILED
startup failed:
/home/vaidotas/tmp/gradle/spock/m5/src/test/groovy/ChildSpec.groovy: 3: Final field 'foo' conflicts with method 'getFoo'; please rename either of them @ line 3, column 5.
       final String foo = "FOO";
```

I do not understand why this worked with m4 in our case.